services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -897,6 +897,22 @@ export class WorkflowService {
       contextHasChanged = !isEqual(data.context, runtimeData.context);
       mergedContext = merge({}, runtimeData.context, data.context);
 
+      const recievedDocumentsMap = keyBy(data.context?.documents ?? [], 'id');
+
+      // This is hack to re-assign properties to documents after merge as temp solution
```

Comment:
We want to tackle tech debt, and we can't add more right now. let's go with the proper solution.

Reply:
@alonp99 I agree with your statement, but I don't think it's correct in this particular case, as it doesn't add more tech debt, it just keeps the existing tech debt. The amount of tech debt remains the same before and after this PR.